src/VirtualPerfCounterMonitor.h

```diff
@@ -18,6 +22,7 @@ namespace rr {
 class VirtualPerfCounterMonitor : public FileMonitor {
 public:
   static bool should_virtualize(const struct perf_event_attr& attr);
+  static VirtualPerfCounterType virtualization_type(const struct perf_event_attr& attr);
```

Comment:
Did you see where I suggested instead of doing this, create a synthetic perf counter ID that we use on all architectures that always counts `ticks`, and use that when running rr replay under rr?

Reply:
Yes, and I think it's a good idea, I just haven't gotten around to it yet. I'd suggest simply not virtualizing performance counters on AMD CPUs until we introduce that.

Reply:
If you don't virtualize at all then the rr under rr test (`record_replay.run`) will fail.